iTechArt.Repositories.EntityFramework/Repository/Repository.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using iTechArt.Repositories.EntityFramework.Interfaces;
+using iTechArt.Repositories.Interfaces;
+
+namespace iTechArt.Repositories.EntityFramework.Repository
+{
+    public class Repository<T> : IRepository<T> where T : class
+    {
+        private readonly IDbContext _dbContext;
+
+        public static Repository<T> Instance;
```

Comment:
Think about another mechanism for caching. It's better to implement this in UnitOfWork(use Dictionary).